Sources/SwiftDocC/Model/Rendering/RenderContentCompiler.swift

```diff
@@ -207,6 +212,44 @@ struct RenderContentCompiler: MarkupVisitor {
         return [RenderInlineContent.strikethrough(inlineContent: strikethrough.children.reduce(into: [], { result, child in result.append(contentsOf: visit(child))}) as! [RenderInlineContent])]
     }
 
+    mutating func visitBlockDirective(_ blockDirective: BlockDirective) -> [RenderContent] {
+        switch blockDirective.name {
+        case Snippet.directiveName:
+            guard let snippetURL = blockDirective.arguments()[Snippet.Semantics.Path.argumentName],
+                  let snippetNode = resolveSymbolReference(destination: snippetURL.value),
+                  let snippetEntity = try? context.entity(with: snippetNode.reference),
+                  let snippetSymbol = snippetEntity.symbol,
+                  let snippetMixin = snippetSymbol.mixins[SymbolGraph.Symbol.Snippet.mixinKey] as? SymbolGraph.Symbol.Snippet else {
+                return []
+            }
+
+            let docCommentContent = snippetEntity.markup.children.flatMap { self.visit($0) }
+
+            let codeContent = snippetMixin.chunks.flatMap { chunk -> [RenderContent] in
+                guard !chunk.code.isEmpty else {
+                    return []
+                }
+
+                var elements = [RenderContent]()
+
+                if let chunkName = chunk.name {
+                    elements.append(RenderBlockContent.paragraph(inlineContent: [
+                        RenderInlineContent.strong(inlineContent: [
+                            RenderInlineContent.text(chunkName)
+                        ])
+                    ]))
+                }
+
+                elements.append(RenderBlockContent.codeListing(syntax: "swift", code: [chunk.code], metadata: nil))
```

Comment:
Does the input data for snippets specify what language they're written in? We're trying to move away as much as possible from hardcoding Swift with the work we're doing to support more languages.

Reply:
Ah yes, I added this to the format after the first revision so will pull that in.